eng/common/pipelines/templates/jobs/generate-job-matrix.yml

```diff
@@ -0,0 +1,139 @@
+parameters:
+- name: AdditionalParameters
+  type: object
+- name: DependsOn
+  type: object
+  default: null
+- name: CloudConfig
+  type: object
+  default: {}
+- name: MatrixConfigs
+  type: object
+  default: []
+- name: MatrixFilters
+  type: object
+  default: []
+- name: MatrixReplace
+  type: object
+  default: {}
+- name: JobTemplatePath
+  type: string
+# Set this to false to do a full checkout for private repositories with the azure pipelines service connection
+- name: SparseCheckout
+  type: boolean
+  default: true
+- name: SparseCheckoutPaths
+  type: object
+  default: []
+- name: Pool
+  type: string
+  default: azsdk-pool-mms-ubuntu-2204-general
+- name: OsVmImage
```

Comment:
Is this still interesting after the migration?

Reply:
Similar question for the Pool parameter.

Reply:
Correct me if I'm wrong @scbedd, but we found demands does not work for the Azure Pipelines mac pool, and we still need to pass `vmImage` for that. Additionally for the migration we'll want to use the main pools and specify the alternate image.

Reply:
> Correct me if I'm wrong @scbedd, but we found demands does not work for the Azure Pipelines mac pool, and we still need to pass `vmImage` for that. Additionally for the migration we'll want to use the main pools and specify the alternate image.

Correct. The specific combo of

```
pool:
  name: 
  vmImage: 
  os: linux|windows|macOS
```

works for both devops + the additional agent checks. I swear I tried everything else.